### ISSUE TITLE
Doesn't work with HASS.IO - Official Component?

### ISSUE THREAD

**Issue by mouth4war** [xiaohuim/homeassistant-magicblue#3]
So I installed Hass.io and this component doesn't load. Can it be made official so this problem doesn't happen?

> 017-08-03 01:06:37 ERROR (SyncWorker_6) [homeassistant.util.package] Unable to install package magicblue==0.4.3: Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-mhr74ky0/bluepy/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-98pxt0j5-record/install-record.txt --single-version-externally-managed --prefix  --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-mhr74ky0/bluepy/
2017-08-03 01:06:37 ERROR (MainThread) [homeassistant.setup] Not initializing light.magicbluelight because could not install dependency magicblue==0.4.3

**Comment by Betree** [comment]
That's strange, what if you install magicblue manually ?

> /srv/homeassistant/bin/pip install magicblue

**Comment by mouth4war** [comment]
It was working fine on HASSbian. I migrated to HASS.io today and get this error. Apparently, HASS.io has similar issues with other custom components also. I've posted in the [HA forums](https://community.home-assistant.io/t/custom-component-dependency-package-install-failure-bluepy-setup-py-error-code-1/23423).

Don't you think this component works good enough to be official? I'm pretty satisfied with it. And since HASS.io seems to be the recommended installation method (these days), it would cater to all those new users as well.

**Comment by Betree** [comment]
It would be nice to make it official but @xiaohuim doesn't seems very active, there haven't been any new commits since 3 months and @StevenLooman's [pull request](https://github.com/xiaohuim/homeassistant-magicblue/pull/1) is waiting since 2 months. 

I'd prefer seeing @StevenLooman's fork as the official component repo.

**Comment by mouth4war** [comment]
Yes! With your commit for effects. Now I need an automation to detect Doom on my PS4 and start red strobe effect :)

@Stevenlooman : let's make it official, shall we?

**Comment by mouth4war** [comment]
So I posted about this on the HASSIO git as well. I've got a reply from the main dev. Looks like the only way this component will work with HASSIO is if it goes official.

https://github.com/home-assistant/hassio/issues/129#issuecomment-319925949

> We support only official components. I refactory the hole image for next release and we support a lot more bluetooth devices. If a custom component depends on a requerement of a official component it will work, it work also if the module is a pure python modul.

> Most of that bluetooth stuff need a complex build system local, that build system use 200-300mb only for compile a 60kb library. We ship the same image to a lot of users and most of it don't need a build system.

> If this component will a part of official home-assistant component or use librarys that is used by other core components we can make that it will work.

> A very big problem is, that the bluetooth module of raspberry pi3 work over the CPU and slow done the hole system. That make that on search the cpu go to slow for home-assistant. So our official statement is, don't use bluetooth on same device as home-assistant run.

**Comment by StevenLooman** [comment]
@mouth4war I'd rather not fork it and continue from that. Better would be that @xiaohuim would merge the pull request. Another possibility is that @xiaohuim makes me co-owner, then I can continue working on this (though, I don't have any further pending changes at the moment.)

**Comment by StevenLooman** [comment]
Perhaps @xiaohuim is also on vacation currently. It is the time... (at least where I live.)

**Comment by Betree** [comment]
@mouth4war Did you solve this issue ? Seems HASS.io uses bluepy 1.1.1, maybe upgrading to the same version in magicblue would do the trick ?

**Comment by tomh05** [comment]
I tried forking [Betree/magicblue](https://github.com/Betree/magicblue), editing it to use bluepy 1.1.1, and then pointing homeassistant-magicblue to my fork by changing the `REQUIREMENTS` variable, but Hassio didn't seem to like loading requirements from urls instead of pip packages.

@xiaohuim doesn't seem to have had any github activity for 4 months. I'd be in favour of a new official fork unless we hear back from him.

**Comment by Betree** [comment]
@tomh05 Thank you for this. I'll start considering the creation of a new repo maintained along with the main project and the release of it as an official HomeAssistant component.

I'll update this thread with additional info when I'll start to work on this.

@xiaohuim Please send us a sign if you come back here before we start moving :wink:

**Comment by tomh05** [comment]
I've managed to get an instance of bluepy installed in as a hassio addon. Perhaps if we don't manage to get bluepy building in homeassistant, a workaround could be to make a 'bluepy bulb' add on (that's a docker container with bluepy and an API), and then a 'smart bulb' component that communicates with that add-on?
Certainly would be simpler to get bluepy working on hassio natively though.

**Comment by Betree** [comment]
As stated on [this comment](https://github.com/home-assistant/hassio/issues/129#issuecomment-319996111) HASS.io now comes with bluepy pre-installed so we should be able to find our way from here.
We can keep your first proposition as a backup plan :wink:

**Comment by tomh05** [comment]
After a discussion over at https://github.com/home-assistant/hassio/issues/129#issuecomment-331624445 I got this working for me.

I changed the requirements of magicblue to use `0.5.0.dev1` and it worked for me (provided I installed and started the bluetooth hassio addon too)

To do it, replace:
```python
REQUIREMENTS = ['magicblue==0.2.3']
```
with
```python
REQUIREMENTS = ['magicblue==0.5.0.dev1']
```

**Comment by Betree** [comment]
That's cool :+1: It also means we'll have to match HassIO's bluepy version in future releases.

I'm in a rush with other projects for now but I'll try to bump on this next month to build a healthy `homeassistant-magicblue` on a new repo ( except if @xiaohuim comes back :octopus: ) and maybe releasing it as an official HA component. @tomh05 If you feel like maintaining it you can take care of this :wink:

**Comment by marcomow** [comment]
Sorry for my newbieness - but how is possible to install this cool component in Hassio? Through ssh it doesn't let me use sudo neither apt-get ...

**Comment by Betree** [comment]
@marcomow we haven't released it yet as an official component for the reasons above. We need to.

I haven't tried it myself, but you should be able to install it using @tomh05's fork https://github.com/tomh05/homeassistant-magicblue.

**Comment by tomh05** [comment]
@marcomow You shouldn't need either:

SSH into your hassio, then copy magicbluelight.py file to :
`<YOUR_CONFIG_DIR>/custom_component/light/magicbluelight.py` Then change the line inside to `REQUIREMENTS = ['magicblue==0.5.0.dev1']` as my post above. It should then automatically install and configure on next reboot.

**Comment by marcomow** [comment]
@Betree @tomh05 thank you for the suggestions! I tried, but doesn't seem to be any sign of life...
- tried " magicblueshell: not found"
- @tomh05 file you refer is this, right? https://github.com/xiaohuim/homeassistant-magicblue/blob/master/magicbluelight.py
- am I missing anything else?

**Comment by tomh05** [comment]
@marcomow I can't remember if magicblueshell works or not, try adding a light to your configuration.yaml as in the README and restarting, and see if one appears on the dashboard. At any rate you should see some more meaningful errors on the developer logs page on the hassio dashboard.

**Comment by marcomow** [comment]
@tomh05 thank you, but not working

here the error
 `
2017-11-15 14:13:23 ERROR (MainThread) [homeassistant.bootstrap] Error loading /config/configuration.yaml: mapping values are not allowed here
  in "/config/configuration.yaml", line 107, column 9
`

**Comment by tomh05** [comment]
That looks like a misformed yaml error rather than a magicblue error. Maybe
you're pasting it in the wrong place, or need to tweak it a bit. Use
http://www.yamllint.com/ if you're still stuck where bug is.

**Comment by marcomow** [comment]
thank you for suggestion! Still...
`  General Errors: 
    - Platform not found: light.magicbluelight
`
any idea? Maybe I missed something in config? What I did:
- I took the file from this repository "magicbluelight.py" and made the edit you suggested
- I put it in /config/custom_component/light/
- I corrected  the sintax (in readme was missing"-" before platform) 
`light:
  -platform: magicbluelight
    name: 'Living Room'
    address: 20:16:01:01:05:a0
    version: 9`

**Comment by marcomow** [comment]
hello! tried again to fix situation, updated to Home Assistant 0.58.1 still having error "2017-11-29 10:00:50 ERROR (MainThread) [homeassistant.loader] Unable to find component light.magicbluelight"

my config
```
light:
  - platform: magicbluelight
    name: 'Living Room'
    address: 20:16:01:01:05:a0
    version: 9
```

previously I took the file from this repository "magicbluelight.py" and made the suggested edit, then I put it in /config/custom_component/light/

ps In case someone managed to take steps forward on custom component, please let know :)

**Comment by marcomow** [comment]
Hello! Does anyone have news regarding custom component? :)

**Comment by Betree** [comment]
Hey @marcomow I'm overwhelmed with work right now and can't work on this. @xiaohuim has not shown any sign of life in more than 6 months so if you want to handle this task by forking the repo and releasing the component I'll be happy to to promote it from magicblue's repo and support you where I can.

**Comment by marcomow** [comment]
Dear @Betree if I only had the skills to do this! Sorry I didn't want to put pressure!

**Comment by Betree** [comment]
No problem 😃 I'd love to see that feature too! It doesn't require a lot of skills, just some time and motivation. If you have both take your chance, we'll help you as we can 😉

**Comment by marcomow** [comment]
He, unfortunately not that skilled yet...so should just have to be patient!
:) Wish you an awesome year!

On Fri, Dec 22, 2017 at 3:54 PM, Benjamin Piouffle <notifications@github.com

**Comment by G1K** [comment]
> ERROR (MainThread) [homeassistant.loader] Unable to find component light.magicbluelight"

fix, components structure changes ha
